### PR TITLE
[3.6] reboot tracking for empty rebootid

### PR DESCRIPTION
**Pull request by kvahed**
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/7846/

### CODE REVIEW

File: arangod/RestHandler/RestReplicationHandler.cpp [hunk 1]
```diff
@@ -3076,10 +3076,13 @@ Result RestReplicationHandler::createBlockingTransaction(
...
-    auto rGuard = std::make_unique<CallbackGuard>(
+
+    std::unique_ptr<CallbackGuard> rGuard = nullptr;
+    if (!serverId.empty()) {
+      rGuard = std::make_unique<CallbackGuard>(
         ci.rebootTracker().callMeOnChange(RebootTracker::PeerState(serverId, rebootId),
                                           f, comment));
-
+    }
     queryRegistry->insert(id, query.get(), ttl, true, true, std::move(rGuard));
```
Comment (joerg84) [inline arangod/RestHandler/RestReplicationHandler.cpp]: what happens if rGuard is nullptr?

Comment (kvahed) [reply arangod/RestHandler/RestReplicationHandler.cpp]: nothing

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7847/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**